src/scripts/main.js

```diff
@@ -1 +1,90 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const clickHandler = () => {
+    clearTimeout(timer);
+    document.removeEventListener('click', clickHandler);
+
+    resolve('First promise was resolved');
+  };
+
+  document.addEventListener('click', clickHandler);
+
+  const timer = setTimeout(() => {
+    document.removeEventListener('click', clickHandler);
+
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+firstPromise
+  .then((message) => {
+    const newElement = document.createElement('div');
+
+    newElement.setAttribute('data-qa', 'notification');
+    newElement.classList.add('success');
+    newElement.innerText = message;
+    document.body.appendChild(newElement);
+  })
+  .catch((error) => {
+    const newElement = document.createElement('div');
+
+    newElement.setAttribute('data-qa', 'notification');
+    newElement.classList.add('error');
+    newElement.innerText = error;
+    document.body.appendChild(newElement);
+  });
+
+const secondPromise = new Promise((resolve, reject) => {
+  const clickHandler = () => {
+    document.removeEventListener('click', clickHandler);
+
+    resolve('Second promise was resolved');
+  };
+
+  document.addEventListener('click', clickHandler);
+  document.addEventListener('contextmenu', clickHandler);
```

Comment:
Adding 'clickHandler' to both 'click' and 'contextmenu' events could lead to unexpected behavior because the same handler will be called for different event types. Consider using separate handlers or checking the event type within the handler.